task10.rb

```diff
@@ -0,0 +1,24 @@
+# Return the maximum product of three numbers in an array, so [-8, -9, 1, 2, 7] outputs 504 (which is -8*-9*7)
+
+def task10(arr)
+  raise ArgumentError, "Not a valid format. Need an array" if arr.class != Array
+  raise ArgumentError, "Array is empty. This is not valid" if arr.empty?
+  raise ArgumentError, "Not a valid format. Need an array with more then two numbers inside" if arr.length < 3 
+  arr.each do |number|
+    raise ArgumentError, "Not a valid format. Elements must be Integer or Float numbers." if number.class != Integer && number.class != Float
+  end
+
+  if arr.length >= 5
```

Comment:
@MaraEfimova Would you remind me why is this sorting needed here?

Reply:
If there are many elements in the array, this will help to avoid time-consuming creation of combinations.

Reply:
So I sorted them and then take only 5 (two biggest from negative numbers and three positive numbers)

Reply:
@MaraEfimova Sorting of long arrays might sometimes be heavier than just calculating a few combinations more, so I would say it really depends. And here you're actually using a "hack". But OK, let's leave it like it is :+1: